joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -157,18 +157,18 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
   /** Attempts to extract a type from the base of a member call.
     */
   protected def typeFromCallTarget(baseNode: RubyExpression): Option[String] = {
-    scope.lookupVariable(baseNode.text) match {
-      // fixme: This should be under type recovery logic
-      case Some(decl: NewLocal) if decl.typeFullName != Defines.Any             => Option(decl.typeFullName)
-      case Some(decl: NewMethodParameterIn) if decl.typeFullName != Defines.Any => Option(decl.typeFullName)
-      case Some(decl: NewLocal) if decl.dynamicTypeHintFullName.nonEmpty => decl.dynamicTypeHintFullName.headOption
-      case Some(decl: NewMethodParameterIn) if decl.dynamicTypeHintFullName.nonEmpty =>
-        decl.dynamicTypeHintFullName.headOption
+    baseNode match {
+      case literal: LiteralExpr => Option(literal.typeFullName)
       case _ =>
-        astForExpression(baseNode).nodes
```

Comment:
Why have we removed this?

Reply:
It can be attributed to quite the performance slowdown for these long chained calls, especially as `base` and `receiver` are often the same long tree for these chained calls. I initially found it to be the cause for generating about 250 TypeRefs that were thrown away, so I think we can simply rely on non-nested nodes to give us the type hint.